### ISSUE TITLE
NoClassDefFoundError whenever using JNPMService.bestMatch()

### ISSUE THREAD

**Issue by Ensar025** [OrienteerBAP/JNPM#15]
Using JNPMService.bestMatch(), as noted in the [example](https://github.com/OrienteerBAP/JNPM#java-api), throws java.lang.NoClassDefFoundError.

Code executed:
```java
var vueVersionInfo = jNpmService.bestMatch("vue@<2");
```

Stack trace:
```red
Exception in thread "main" java.lang.NoClassDefFoundError: javax/activation/MimetypesFileTypeMap
	at org.orienteer.jnpm.JNPMUtils.<clinit>(JNPMUtils.java:213)
	at org.orienteer.jnpm.RxJNPMService.retrieveVersions(RxJNPMService.java:71)
	at java.base/java.lang.invoke.MethodHandle.invokeWithArguments(MethodHandle.java:731)
	at retrofit2.Platform.invokeDefaultMethod(Platform.java:109)
	at retrofit2.Retrofit$1.invoke(Retrofit.java:159)
	at jdk.proxy1/jdk.proxy1.$Proxy0.retrieveVersions(Unknown Source)
	at org.orienteer.jnpm.RxJNPMService.retrieveVersions(RxJNPMService.java:89)
	at java.base/java.lang.invoke.MethodHandle.invokeWithArguments(MethodHandle.java:731)
	at retrofit2.Platform.invokeDefaultMethod(Platform.java:109)
	at retrofit2.Retrofit$1.invoke(Retrofit.java:159)
	at jdk.proxy1/jdk.proxy1.$Proxy0.retrieveVersions(Unknown Source)
	at org.orienteer.jnpm.RxJNPMService.bestMatch(RxJNPMService.java:100)
	at java.base/java.lang.invoke.MethodHandle.invokeWithArguments(MethodHandle.java:731)
	at retrofit2.Platform.invokeDefaultMethod(Platform.java:109)
	at retrofit2.Retrofit$1.invoke(Retrofit.java:159)
	at jdk.proxy1/jdk.proxy1.$Proxy0.bestMatch(Unknown Source)
	at org.orienteer.jnpm.JNPMService.bestMatch(JNPMService.java:139)
	at com.io.jnpmfetch.fetcher.NPMFetcher.fetchDependencies(NPMFetcher.java:56)
	at com.io.jnpmfetch.fetcher.NPMFetcher.fetchForDepth(NPMFetcher.java:41)
	at com.io.jnpmfetch.Main.main(Main.java:9)
Caused by: java.lang.ClassNotFoundException: javax.activation.MimetypesFileTypeMap
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:641)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:188)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:521)
	... 20 more
```

The issue seems to be with JNPMUtils.toVersionPredicate(versionConstraint). Importing the package:
```xml
        <dependency>
            <groupId>javax.activation</groupId>
            <artifactId>activation</artifactId>
            <version>1.1.1</version>
        </dependency>
```
fixed the problem but this isn't noted anywhere in the README.md file. Is this behaviour within expectations?

**Comment by PhantomYdn** [comment]
@Ensar025 , thank you for reporting! Seems that you are using Java11 - and, as far as I can see: there are changes in modules that are available. I will refactor this place to avoid such issues with modern java versions

**Comment by PhantomYdn** [comment]
@Ensar025, issue was fixed in version 1.4
